### DISCUSSION TITLE
Understanding Kornia Augmentation Pipeline and which transformations were applied

### DISCUSSION THREAD

**Discussion by sohamb91** [kornia/kornia#3191] [Questions] [answered]
I have a created a Kornia augmentation pipeline for training that looks like:

```
self.train_augments = AugmentationSequential(
               K. VideoSequential(
                    Normalize255(),
                    K.ColorJitter(0.3, 0.3, 0.3, 0.1, p=0.8),
                    K.Resize(resize_to),
                    K.RandomRotation(degrees=12.0, p=0.5, same_on_batch=True),
                    K.CenterCrop(resize_to),
                    K.RandomHorizontalFlip(p=0.5, same_on_batch=True),
                    K.Normalize(mean, std),
                    same_on_frame=True
                   
                ),
                data_keys=["input", "keypoints"],
            )

``` 

Normalize255 is a function that i have implemented myself because of pickling errors. it subclasses nn.Module and basically normalizes the image by a factor of 255.0

My question is: in this pipeline, since RandomHorizontalFlip happens with a probability of 0.5, how do i know in which cases the flip augmentation has been applied? i ask because, if flip has been applied, i will need to update my keypoints using flip-indices.

**Comment by edgarriba** [comment]
@sohamb91 i believe the augmentation sequential it has a member that cache with a boolean mask indicating the elements that were triggered. @shijianjian

**Reply by sohamb91** [reply, marked as answer]
yeah. after a pretty deep research session, i managed to find it here like so:
for module in list(augments.VideoSequential_0.children()):
                if isinstance(module, RandomHorizontalFlip):
                    is_batch_flipped = torch.all(module._params["batch_prob"] == 1.0)
